int32 mode
int32 ID
---
int32 mode
int32 ID
bool success
